doc/tools/docgram/doc_grammar.ml

```diff
@@ -417,21 +417,46 @@ let add_symdef nt file symdef_map =
   in
   symdef_map := StringMap.add nt (Filename.basename file::ent) !symdef_map
 
-let rec edit_SELF nt cur_level next_level right_assoc prod =
+let rec edit_SELF nt cur_level next_level right_assoc inner prod =
+  let subedit sym = List.hd (edit_SELF nt cur_level next_level right_assoc true [sym]) in
   let len = List.length prod in
   List.mapi (fun i sym ->
     match sym with
-    | Snterm s -> begin match s with
-      | s when s = nt || s = "SELF" ->
-        if i = 0 then Snterm next_level
-        else if i+1 < len then sym
-        else if right_assoc then Snterm cur_level else Snterm next_level
-      | "NEXT" -> Snterm next_level
-      | _ -> sym
-    end
-    | Slist1 sym -> Slist1 (List.hd (edit_SELF nt cur_level next_level right_assoc [sym]))
-    | Slist0 sym -> Slist0 (List.hd (edit_SELF nt cur_level next_level right_assoc [sym]))
-    | x -> x)
+    | Sterm _ -> sym
+    | Snterm s ->
+      begin match s with
+        | s when s = "SELF" || s = nt ->
+          (* See https://camlp5.github.io/doc/pdf/camlp5-7.00.pdf version 7.00,
+            June 26, 2017 section 8.6.2:
+
+            This seems correct:
+            "When a SELF or the current entry name is encountered in the middle of the rule
+             (i.e. if it is not the last symbol), there is a call to the "start" function
+             of the First level of the current entry."
+
+            but this is incorrect, contradicting the first statement:
+            "The left "SELF"s of the two levels "minus" and "power" correspond to a call to the next level."
+
+            Examples:
+            once idtac + (once idtac).  ACCEPTED -> lhs goes to top
+            idtac + once idtac.  PARSE ERROR -> rhs is not to top level, sb at current level
+          *) 
```

Comment:
This is what makes the linter complain. Should be squashed to make it happy.

```suggestion
          *)
```

Reply:
The hook for removing whitespace runs whenever I commit.  I would have expected it would have fixed this.

Reply:
Yeah, me too.